ggml/src/CMakeLists.txt

```diff
@@ -213,7 +213,7 @@ add_library(ggml
 
 target_link_libraries(ggml PUBLIC ggml-base)
 
-if (CMAKE_SYSTEM_NAME MATCHES "Linux")
```

Comment:
Should there be a special branch for https://releases.llvm.org/10.0.0/projects/libcxx/docs/UsingLibcxx.html#id2?

Reply:
Also idk about `libdl` handling

Reply:
> GCC on RHEL 8,

Let's then add a comment explaining the "why"s